Source/MonoGame.Extended/Shapes/Circle.cs

```diff
@@ -0,0 +1,506 @@
+		/// </summary>
+		/// <param name="value">Other <see cref="Circle"/>.</param>
+		/// <returns><c>true</c> if other <see cref="Circle"/> intersects with this <see cref="Circle"/>; <c>false</c> otherwise.</returns>
+		public bool Intersects(Circle value)
+		{
+			Vector2 distanceOfCenter = value.Center - Center;
+			float radii = Radius + value.Radius;
+
+			return ((distanceOfCenter.X * distanceOfCenter.X) + (distanceOfCenter.Y * distanceOfCenter.Y) < (radii * radii));
+		}
+
+		/// <summary>
+		/// Gets whether or not a specified <see cref="Circle"/> intersects with this <see cref="Circle"/>.
+		/// </summary>
+		/// <param name="value">Other <see cref="Circle"/>.</param>
+		/// <param name="result"><c>true</c> if other <see cref="Circle"/> intersects with this <see cref="Circle"/>; <c>false</c> otherwise. As an output parameter.</param>
+		public void Intersects(ref Circle value, out bool result)
+		{
+			Vector2 distanceOfCenter = value.Center - Center;
+			float radii = Radius + value.Radius;
+
+			result = ((distanceOfCenter.X * distanceOfCenter.X) + (distanceOfCenter.Y * distanceOfCenter.Y) < (radii * radii));
+		}
+
+		/// <summary>
+		/// Gets whether or not a specified <see cref="Rectangle"/> intersects with this <see cref="Circle"/>.
+		/// </summary>
+		/// <param name="value">Other <see cref="Rectangle"/>.</param>
+		/// <returns><c>true</c> if other <see cref="Rectangle"/> intersects with this <see cref="Circle"/>; <c>false</c> otherwise.</returns>
+		public bool Intersects(Rectangle value)
+		{
+			Vector2 distance = new Vector2(Math.Abs(Center.X - value.X), Math.Abs(Center.Y - value.Y));
+
+			if (distance.X > (value.Width / 2.0f + Radius))
+				return false;
+			if (distance.Y > (value.Height / 2.0f + Radius))
+				return false;
+
+			if (distance.X <= (value.Width / 2.0f))
+				return true;
+			if (distance.Y <= (value.Height / 2.0f))
+				return true;
+
+
+			float distanceOfCorners = ((distance.X - value.Width / 2.0f) * (distance.X - value.Width / 2.0f) +
+									   (distance.Y - value.Height / 2.0f) * (distance.Y - value.Height / 2.0f));
+
+			return (distanceOfCorners <= (Radius * Radius));
+		}
+
+		/// <summary>
+		/// Gets whether or not a specified <see cref="Rectangle"/> intersects with this <see cref="Circle"/>.
+		/// </summary>
+		/// <param name="value">Other <see cref="Rectangle"/>.</param>
+		/// <param name="result"><c>true</c> if other <see cref="Rectangle"/> intersects with this <see cref="Circle"/>; <c>false</c> otherwise. As an output parameter.</param>
+		public void Intersects(ref Rectangle value, out bool result)
+		{
+			Vector2 distance = new Vector2(Math.Abs(Center.X - value.X), Math.Abs(Center.Y - value.Y));
+
+			if (distance.X > (value.Width / 2.0f + Radius))
+				result = false;
+			if (distance.Y > (value.Height / 2.0f + Radius))
+				result = false;
+
+			if (distance.X <= (value.Width / 2.0f))
+				result = true;
+			if (distance.Y <= (value.Height / 2.0f))
+				result = true;
```

Comment:
These `result = true` lines look like bugs? The result *always* gets overridden at the end of the method so the if statements are not doing anything useful.